nilearn/input_data/nifti_masker.py

```diff
@@ -265,8 +269,36 @@ def fit(self, imgs=None, y=None):
         self.mask_img_.get_data()
         if self.verbose > 10:
             print("[%s.fit] Finished fit" % self.__class__.__name__)
+
+        # Optionally retain input data for report generation
+        if self.reports:
+            dim = image.load_img(imgs).shape
+            img = image.index_img(imgs, dim[-1] // 2)
+            self.input_ = img
         return self
 
+    def generate_report(self):
+        """Generates a report for NiftiMasker
+
+        Returns
+        -------
+        report : HTMLDocument
+        """
+        if not hasattr(self, 'input_'):
+            warnings.warn('Report generation not enabled !'
+                          'No visual outputs will be created.')
+            return
```

Comment:
if we want to return consistent types, when this problem occurs reported objects could return an HTMLDocument that contains an error message explaining the problem. but maybe this is not useful

Reply:
I updated it to give an HTML document -- right now it doesn't explain the problem, other than that reports were not generated. Would be happy to think on how this could be done better !